metricflow/test/model/modify/modify_metric_filter.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from dbt_semantic_interfaces.implementations.filters.where_filter import PydanticWhereFilterIntersection
+from dbt_semantic_interfaces.implementations.semantic_manifest import PydanticSemanticManifest
+from dbt_semantic_interfaces.references import MetricReference
+from dbt_semantic_interfaces.transformations.transform_rule import SemanticManifestTransformRule
+from typing_extensions import override
+
+
+class ModifyMetricFilterTransform(SemanticManifestTransformRule[PydanticSemanticManifest]):
+    """Modifies the specified metric to have the given filter.
+
+    This is useful for programmatically generally different manifests to create different test cases.
```

Comment:
There's a bit of a repeat there, but if you only look at 1 file, you might wonder why it's useful. It would be good to centralize it though. Maybe an interface.